### PR TITLE
Kwesi/net-1106-allow-setting-connection-endpoint-on

### PR DESCRIPTION
**Pull request by Aceix**
## Describe your changes
this PR enables a user to add additional endpoints for an RAG, and select ehich to use when connecting with RAC

## Checklist before requesting a review
- [ ] My changes affect only 10 files or less.
- [ ] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netmaker is awesome.

### CODE REVIEW

File: controllers/ext_client.go [hunk 1]
```diff
@@ -200,6 +201,23 @@ func getExtClientConf(w http.ResponseWriter, r *http.Request) {
...
+	if preferredIp != "" {
+		allowedPreferredIps := []string{}
+		for i := range gwnode.AdditionalRagIps {
+			allowedPreferredIps = append(allowedPreferredIps, gwnode.AdditionalRagIps[i].String())
+		}
+		allowedPreferredIps = append(allowedPreferredIps, host.EndpointIP.String())
+		allowedPreferredIps = append(allowedPreferredIps, host.EndpointIPv6.String())
+		if !slices.Contains(allowedPreferredIps, preferredIp) {
+			slog.Warn("preferred endpoint ip is not associated with the RAG. proceeding with preferred ip", "preferred ip", preferredIp)
+			return
```
Comment (abhishek9686) [inline controllers/ext_client.go:214]: we are returning here with out writing api response

File: controllers/ext_client.go [hunk 2]
```diff
@@ -215,12 +233,18 @@ func getExtClientConf(w http.ResponseWriter, r *http.Request) {
...
 	}
+
 	gwendpoint := ""
-	if host.EndpointIP.To4() == nil {
-		gwendpoint = fmt.Sprintf("[%s]:%d", host.EndpointIP.String(), host.ListenPort)
+	if preferredIp == "" {
+		if host.EndpointIP.To4() == nil {
+			gwendpoint = fmt.Sprintf("[%s]:%d", host.EndpointIPv6.String(), host.ListenPort)
+		} else {
+			gwendpoint = fmt.Sprintf("%s:%d", host.EndpointIP.String(), host.ListenPort)
```
Comment (abhishek9686) [inline controllers/ext_client.go:242]: default preferred endpoint should be ipv4

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: controllers/ext_client.go (hunk 2 above)
Comment (Aceix) [reply controllers/ext_client.go:242]: ipv4 is the preferred endpoint

File: controllers/ext_client.go (hunk 1 above)
Comment (Aceix) [reply controllers/ext_client.go:214]: addressed

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**